client/src/routes/PageSearch.js

```diff
@@ -18,6 +18,36 @@ import SearchEasterEggs from '../utilities/SearchEasterEggs';
 
 
 class PageSearch extends React.Component {
+  async componentDidMount() {
+    const p = this.props;
+
+    if(window.location.search.length !== 0) {
+      const urlParams = new URLSearchParams(window.location.search);
+
+      if (!p.search.filterOptions) {
+        await p.dispatch(fetchFilterOptions())
+      }
+
+      for(const [key, value] of urlParams.entries()) {
+        if(key !== 'query') {
+          if(key === "category" || key === "subcategory" || key === "tournament") {
+            p.dispatch(updateSearchFilter(key, value.split(',').map(e => parseInt(e, 10))))
+          } else {
+            p.dispatch(updateSearchFilter(key, value.split(',')))
+          }
+        } else {
+          p.dispatch(updateSearch(value))
+        }
+      }
+
+      SearchEasterEggs(this.props.dispatch, this.props.search.query);
+      p.dispatch(fetchQuestions({
+        searchQuery: this.props.search.query,
+        searchFilters: this.props.search.filters
+      }));
```

Comment:
Likewise, it's probably good to extract this two func block, given that it's the exact same as in the `onSearch()` func

Reply:
Done 👍